src/main/java/io/gravitee/common/util/VertxProxyOptionsUtils.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.common.util;
+
+import io.vertx.core.net.ProxyOptions;
+import io.vertx.core.net.ProxyType;
+import java.util.Objects;
+import org.springframework.context.ApplicationContext;
+import org.springframework.core.env.Environment;
+
+/**
+ * @author GraviteeSource Team
+ */
+public class VertxProxyOptionsUtils {
+
+    static final String PROXY_HOST_PROPERTY = "system.proxy.host";
+    static final String PROXY_PORT_PROPERTY = "system.proxy.port";
+    static final String PROXY_TYPE_PROPERTY = "system.proxy.type";
+    static final String PROXY_USERNAME_PROPERTY = "system.proxy.username";
+    static final String PROXY_PASSWORD_PROPERTY = "system.proxy.password";
+
+    public static ProxyOptions initFromSpringContext(ApplicationContext applicationContext) {
+        final Environment environment = applicationContext.getEnvironment();
+        final ProxyOptions proxyOptions = new ProxyOptions();
+        try {
+            proxyOptions.setHost(environment.getProperty(PROXY_HOST_PROPERTY));
+            proxyOptions.setPort(parseProxyPort(environment.getProperty(PROXY_PORT_PROPERTY)));
+            proxyOptions.setType(ProxyType.valueOf(environment.getProperty(PROXY_TYPE_PROPERTY)));
```

Comment:
Could it be possible to keep the same logic we had previously: check host, port, type, collect the errors into a builder and throw the exception at the end ? 
The idea behind this is to give a complete list of all the problem you have with your proxy configuration. This avoid the user to fix one problem, restart, see another problem, fix it, restart, see another problem and so on...